nomad/data_source_acl_tokens.go

```diff
@@ -0,0 +1,83 @@
+package nomad
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceACLTokens() *schema.Resource {
+	return &schema.Resource{
+		Read: aclTokensDataSourceRead,
+
+		Schema: map[string]*schema.Schema{
+			"prefix": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"acl_tokens": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"accessor_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"policies": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"global": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"create_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func aclTokensDataSourceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(ProviderConfig).client
+
+	tokens, _, err := client.ACLTokens().List(nil)
```

Comment:
`List()` can be passed a `QueryOptions` including a prefix:
https://github.com/hashicorp/nomad/blob/dc6aa33916faf4e6129e138f483e0e6b88c975b7/api/acl.go#L86
https://github.com/hashicorp/nomad/blob/dc6aa33916faf4e6129e138f483e0e6b88c975b7/api/api.go#L59

This will reduce the size of the response and avoid the need for filtering below.

Reply:
Thanks!